sdks/java/extensions/avro/src/main/java/org/apache/beam/sdk/extensions/avro/schemas/utils/AvroUtils.java

```diff
@@ -441,16 +441,35 @@ public TypeConversion<StackManipulation> createSetterConversions(StackManipulati
   }
 
   /** Get Beam Field from avro Field. */
-  public static Field toBeamField(org.apache.avro.Schema.Field field) {
-    TypeWithNullability nullableType = new TypeWithNullability(field.schema());
+  public static Schema.Field toBeamField(org.apache.avro.Schema.Field field) {
+    org.apache.avro.Schema avroSchema = field.schema();
+    TypeWithNullability nullableType = TypeWithNullability.create(avroSchema);
     FieldType beamFieldType = toFieldType(nullableType);
-    return Field.of(field.name(), beamFieldType);
+
+    if (beamFieldType.getTypeName() == TypeName.DECIMAL) {
```

Comment:
This adds a special case to `toBeamField`. Appreciate if add a brief comment for the purpose?

Reply:
As far as I understood, here we extract aAvro type properties and brings them into beam's type options.